executor/predictor/predictor_process.go

```diff
@@ -329,7 +336,17 @@ func (p *PredictorProcess) predictChildren(node *v1.PredictiveUnit, msg payload.
 				return cmsgs[0], err
 			}
 		}
-		return p.aggregate(node, cmsgs, msg, puid)
+		amsg, err := p.aggregate(node, cmsgs, msg, puid)
```

Comment:
As `aggregate` also calls `p.logPayload` would't some messages be logged twice?

Reply:
I think the idea here is you can log the payload after every transformation. Normally, each node in the graph only does one thing like route or combine. But its true you can have a node that does multiple things in which case you would get payloads logged after each operation. The code in `aggregate`only logs if the operation is run.